src/consensus/sync_queue.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <libp2p/Host.h>
+
+#include <deque>
+
+namespace taraxa {
+
+class SyncBlock;
+
+class SyncBlockQueue {
+ public:
+  SyncBlockQueue() = default;
+
+  bool push(SyncBlock &&block, dev::p2p::NodeID const &node_id, uint64_t max_pbft_size);
+  std::pair<SyncBlock, dev::p2p::NodeID> pop();
+  void clear();
+  size_t size() const;
+  uint64_t getPeriod() const;
+
+ private:
+  std::deque<std::pair<SyncBlock, dev::p2p::NodeID>> sync_queue_;
+  // We need this variable as for small amount of time block is not part of queue but still beeing processed
```

Comment:
I would also mention that sync_period_ should be always equal to the sync_queue_.back()->getPeriod(). You forgot about it in pop() function for example